test/run.sh

```diff
@@ -34,10 +34,12 @@ function runCommandAndCompareToSnapshot {
       exit 1
     fi
 
-    eval "$LOCAL_COMMAND$AUTH --FOR-TESTS $ARGS" 2>&1 \
+    (eval "$LOCAL_COMMAND$AUTH --FOR-TESTS $ARGS" || true) 2>&1 \
         | $REPLACE_SCRIPT \
         > "$TMP/$FILE"
-    if [ "$(diff "$TMP/$FILE" "$SNAPSHOTS/$FILE")" != "" ]
+    local diffed
+    diffed="$(diff "$TMP/$FILE" "$SNAPSHOTS/$FILE" || true)"
+    if [ "$diffed" != "" ]
```

Comment:
This is something OSH's `strict_errexit` caught. Before, if diff failed, it would be ignored. That's not good, so it has to be extracted into a variable.
However, we do want to do our own handling of it, so we then explicitly ignore the potential failures.